### DISCUSSION TITLE
bypass-lan plugin

### DISCUSSION THREAD

**Discussion by robdyck** [strongswan/strongswan#1853] [General]
This  may be a Fedora only problem. For years I have been using a work-around that allows a road warrior to access the LAN associated with the strongswan server. bypass-lan.conf does not alter the operation of this plugin in any way. I have tried every combination of load, interfaces_use and interfaces_ignore except listing my interface for both. My workaround is to hide the plugins .so file. Since this works for me it seems the plugin is always loaded regardless of bypass-lan.conf and it always bypasses the LAN.

This is inconvenient when strongswan gets updated. I have to remember to hide the plugin once again. For a while I would build my own strongswan without bypass-lan. This also is inconvenient.

Is there a more elegant way to allow access to the server's LAN.

**Comment by tobiasbrunner** [comment]
> bypass-lan.conf does not alter the operation of this plugin in any way.

Depends on your strongswan.conf. The snippets are only effective if they are included in it (see the example [here](https://docs.strongswan.org/docs/5.9/config/strongswanDir.html)). But you can also just configure everything directly in strongswan.conf e.g. disable the plugin:

    charon {
        plugins {
             bypass-lan {
                  load = no
             }
        }
    }

> Is there a more elegant way to allow access to the server's LAN.

Isn't that exactly what the _bypass-lan_ plugin does?

**Comment by robdyck** [comment]
I tried your suggestion and it worked. I un-hid the binary and retested ping to the servers LAN. For some reason modifying strongswan.d/charon/bypass-lan.conf had no effect. Modular is set to yes and bypass-lan.conf is set to load = no.
I had also tried it with bypass-lan.conf set to load = yes and then adding the subnet. I tried adding to ignore and also include just in case I had the logic backward. but it had no effect. bypass-lan.conf is simply ignored.

It would seem that the purpose of bypass-lan is to exclude the LAN as a destination. When I ensured that the plugin wouldn't load by hiding the binary then the road warrior can ping the server's LAN.

**Comment by shopon16** [comment]
192.168.220.0/24 subnet is my server's LAN. 
Make sure you have enabled bypass-lan during installation.
Here is the configuration to allow your specific subnet. 
```
bypass-lan {
    interfaces_use = 192.168.220.0/24
    load = yes
}
```

**Reply by tobiasbrunner** [reply]
Note that that's not how that configuration options works. It takes the name of an interface, not a subnet (you could define your own bypass policies to exclude specific subnets).

**Comment by robdyck** [comment]
How to interpret that.
bypass the lan except don't bypass 192.168.220.0/24?

The goal is to allow access to the LAN. I hid the plugin to stop bypassing the LAN.
